server/src/test/java/keywhiz/service/daos/GroupDAOTest.java

```diff
@@ -147,6 +153,23 @@ public class GroupDAOTest {
     assertNull(afterWithoutOwner.getOwner());
   }
 
+  @Test public void deleteSetsClientOwnerToNull() {
+    String ownerName = randomName();
+    long ownerId = createGroup(ownerName, NO_OWNER);
+
+    String clientName = randomName();
+    long clientId = createClient(clientName, ownerId);
+
+    Client beforeWithOwner = getClientById(clientId);
+    assertEquals(ownerName, beforeWithOwner.getOwner());
+
+    Group owner = getGroupById(ownerId);
+    groupDAO.deleteGroup(owner);
+
+    Client afterWithoutOwner = getClientById(clientId);
+    assertNull(afterWithoutOwner.getOwner());
+  }
```

Comment:
It turns out that this test is unnecessary. Because we do a left join on the owning group and then only populate the owner name if the owner is present, a "dangling" owner ID field doesn't cause a problem. Of course, that's bad, because it masks an underlying issue. I think the solution is to add a sanity check in the record conversion code that will barf if the owner ID is present but the owning group is not.